### PR TITLE
Alter the way cf_writer handle hardcoded global attributes

### LINKED ISSUE #1191: cf_writer should append to Convention global attribute if given header_attr
Some times user want to add more than the CF convention to the global attribute `Convention`

If the user specify eg. the ACDD (http://wiki.esipfed.org/index.php/Attribute_Convention_for_Data_Discovery) in addition to the CF the Convention is overwritten here:
https://github.com/pytroll/satpy/blob/master/satpy/writers/cf_writer.py#L617

Is it possible to append to the `Convention` global attribute like this
```
if `Conventions` in root.attrs:
  root.attrs['Conventions'] += ', ' + CF_VERSION
else:
  root.attrs['Conventions'] = CF_VERSION
```

### PR DESCRIPTION
**Pull request by TAlonglong**
If the user provide Conventions and/or history this should take precedens over the hardcoded values for this.

This PR applies this.

 - [x] Closes #1191 
 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy`` 
 - [x] Fully documented

### CODE REVIEW

**Comment by mraspaud** [comment]
Thanks for the PR. Can you add a tiny test for this ?

**Comment by TAlonglong** [comment]
That is my plan yes, but I wanted to get, how to say, approval of my solution before I do this.

Also I see that CodeFactor is complaining so I have to do some refactor also.

File: satpy/writers/cf_writer.py [hunk 1]
```diff
@@ -610,10 +610,12 @@ def save_datasets(self, datasets, filename=None, groups=None, header_attrs=None,
             if flatten_attrs:
                 header_attrs = flatten_dict(header_attrs)
             root.attrs = encode_attrs_nc(header_attrs)
-        root.attrs['history'] = 'Created by pytroll/satpy on {}'.format(datetime.utcnow())
+        if 'history' not in root.attrs:
+            root.attrs['history'] = 'Created by pytroll/satpy on {}'.format(datetime.utcnow())
```
Comment (djhoese) [inline satpy/writers/cf_writer.py:614]: This is one case where I wonder if appending is the right thing to do. The history is meant to be a history of how the file was created, right? So maybe appending something to the history would be better than overwriting it? I don't have strong feelings or experience with this though.

Comment (mraspaud) [reply satpy/writers/cf_writer.py:614]: I agree with @djhoese . Here is the relevant snippet from the CF conventions document:
```
history

    Provides an audit trail for modifications to the original data. Well-behaved generic netCDF filters will automatically append their name and the parameters with which they were invoked to the global history attribute of an input netCDF file. We recommend that each line begin with a timestamp indicating the date and time of day that the program was executed.
```

Comment (TAlonglong) [reply satpy/writers/cf_writer.py:614]: OK. But you agree that if the user provide a history in `header_attrs` this should be appended to the hard coded _created by ..._ statement. But we rely on the user to provide this `history` attribute in the recommended format.

**Comment by TAlonglong** [comment]
codebeat beat me again. 4 lines too many ...

**Comment by TAlonglong** [comment]
Hm codefactor happy, but codebeat is worse

I thought a namespace is a like a function. Where is codebeat complaining about 156 lines? I cant figure out.

**Comment by TAlonglong** [comment]
OK so python namespace is classes.
Then is commit is just making things worse. Lets roll back

**Pull request merged**